Content.Server/_NF/M_Emp/M_EmpSystem.cs

```diff
@@ -0,0 +1,351 @@
+                    throw new ArgumentOutOfRangeException();
+            }
+        }
+
+        private void OnInteractHand(EntityUid uid, M_EmpGeneratorComponent component, InteractHandEvent args)
+        {
+            if (args.Handled)
+                return;
+            args.Handled = true;
+            StartGenerator(uid, component, args.User);
+            UpdateAppearance(uid, component);
+        }
+
+        private void OnInit(EntityUid uid, SharedM_EmpComponent component, ComponentInit args)
+        {
+            _linker.EnsureSinkPorts(uid, component.ReceiverPort);
+        }
+
+        private void StartGenerator(EntityUid uid, M_EmpGeneratorComponent component, EntityUid user)
+        {
+            switch (component.GeneratorState.StateType)
+            {
+                case GeneratorStateType.Inactive:
+                    ShowPopup(uid, "m_emp-system-report-activate-success", user);
+                    var generatorTransform = Transform(uid);
+                    var gridId = generatorTransform.GridUid ?? throw new InvalidOperationException("Generator had no grid associated");
+                    if (!_M_EmpGridStates.TryGetValue(gridId, out var gridState))
+                    {
+                        gridState = new M_EmpGridState();
+                        _M_EmpGridStates[gridId] = gridState;
+                    }
+                    gridState.ActiveGenerators.Add(uid);
+
+                    PlayActivatedSound(uid, component);
+
+                    component.GeneratorState = new GeneratorState(GeneratorStateType.Activating, gridState.CurrentTime + component.ActivatingTime);
+                    RaiseLocalEvent(new M_EmpGeneratorActivatedEvent(uid));
+
+                    //Report(uid, component.M_EmpChannel, "m_emp-system-report-activate-success"); Removed to lower spam
+
+                    break;
+                case GeneratorStateType.Activating:
+                case GeneratorStateType.Engaged:
+                    ShowPopup(uid, "m_emp-system-report-already-active", user);
+                    break;
+                case GeneratorStateType.CoolingDown:
+                case GeneratorStateType.Recharging:
+                    ShowPopup(uid, "m_emp-system-report-recharging", user);
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+        }
+        private void ShowPopup(EntityUid uid, string messageKey, EntityUid user)
+        {
+            _popupSystem.PopupEntity(Loc.GetString(messageKey), uid, user);
+        }
+
+        private bool SpawnM_Emp(EntityUid uid, M_EmpGeneratorComponent component)
+        {
+            var station = _station.GetOwningStation(uid);
+            var stationName = station is null ? null : Name(station.Value);
+
+            Report(uid, component.M_EmpChannel, "m_emp-system-announcement-active", ("timeLeft", component.EngagedTime.TotalSeconds), ("grid", stationName!));
```

Comment:
You were passing it a variable `stationName` that _could_ have been null, when your `Report` function is looking for a variable in that parameter that is definitely _not_ null.  Since your report functions without sending anything as an object there for a second parameter, we can use `!` to only send this as an argument if it is _not_ null, otherwise sending no object at all if it is null.

Reply:
Seem to be fine now, thanks.